### PR TITLE
refactor: Restructure the ONNX and mlpack plugins

### PR DESCRIPTION
**Pull request by paulgessinger**
Fixes the CMake configuration, python bindings, naming conventions, build dependencies etc. Should not change anything about the execution itself.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -137,7 +137,8 @@ set_option_if(
   ACTS_BUILD_EXAMPLES OR ACTS_BUILD_EVERYTHING)
 set_option_if(ACTS_BUILD_PLUGIN_LEGACY ACTS_BUILD_EVERYTHING)
 set_option_if(ACTS_BUILD_PLUGIN_AUTODIFF ACTS_BUILD_EVERYTHING)
-set_option_if(ACTS_BUILD_EXAMPLES_EXATRKX ACTS_BUILD_PLUGIN_EXATRKX)
+set_option_if(ACTS_BUILD_PLUGIN_EXATRKX ACTS_BUILD_EXAMPLES_EXATRKX)
+set_option_if(ACTS_BUILD_PLUGIN_ONNX ACTS_BUILD_PLUGIN_MLPACK)
```
Comment (Corentin-Allaire) [inline CMakeLists.txt:141]: Technically MLPACK is fully independent from ONNX and one could be used without the other. It is just that right now the only example in which it is used need both

File: Examples/Python/src/Mlpack.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+namespace Acts::Python {
+
+void addMlpack(Context& ctx) {
+  auto [m, mex, onnx] = ctx.get("main", "examples", "onnx");
+  auto mlpack = mex.def_submodule("_mlpack");
+
+  ACTS_PYTHON_DECLARE_ALGORITHM(
+      ActsExamples::AmbiguityResolutionMLDBScanAlgorithm, mlpack,
+      "AmbiguityResolutionMLDBScanAlgorithm", inputTracks, inputDuplicateNN,
+      outputTracks, nMeasurementsMin, epsilonDBScan, minPointsDBScan);
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply Examples/Python/src/Mlpack.cpp:29]: For G4/DD4hep I added a separate component for this. I think we should do the same here, ONNX, mlpack and ONNX+mlpack with separate `add*` functions, stubs and python modules.

[the remark this answers is not in the document]
Comment (paulgessinger) [reply Examples/Python/src/Mlpack.cpp]: Ok, I call this one `OnnxMlpack` now, and we can add an independent `Mlpack.cpp` down the line if need be.

[the remark this answers is not in the document]
Comment (Corentin-Allaire) [reply Examples/Python/src/Mlpack.cpp]: Thanks ! I agree this is probably the best way to do this !

**Review by Corentin-Allaire — APPROVED** [review]
Thanks a lot for your work ! Maybe we should have ping you directly on the previous PR before merging it...

**Comment by benjaminhuth** [comment]
there is something going wrong with the ExaTrkX test on the CI bridge... undefined symbol `Acts::Python::addOnnxMlpack(Acts::Python::Context&)`.

**Comment by paulgessinger** [comment]
> there is something going wrong with the ExaTrkX test on the CI bridge... undefined symbol `Acts::Python::addOnnxMlpack(Acts::Python::Context&)`.

Right I think that's one missing branch where I don't add the stub. Hope it's fixed now.

**Comment by paulgessinger** [comment]
I think it's fixed now @benjaminhuth

**Review by benjaminhuth — APPROVED** [review]

**Pull request merged**